vespa-cloud/album-recommendation-docproc/src/main/java/ai/vespa/example/album/LyricsDocumentProcessor.java

```diff
@@ -45,13 +43,11 @@ public void handleResponse(Response response) {
                 long reqId = response.getRequestId();
                 if (response instanceof DocumentResponse) {
                     logger.info("  Async response to put or get, requestID: " + reqId);
-                    Processing processing = processings.remove(reqId);
-                    processing.removeVariable(VAR_REQ_ID);
                     DocumentResponse resp = (DocumentResponse)response;
                     Document doc = resp.getDocument();
                     if (doc != null) {
```

Comment:
You should add the response directly here and do the error handling in the process method instead.
If not you will end up with some processings that will never complete because the request failed.